### PR TITLE
magicsock, wgengine, ipn, controlclient: plumb regular netchecks to map request

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: control/controlclient/direct.go [hunk 1]
```diff
@@ -176,6 +176,21 @@ func (c *Direct) SetHostinfo(hi *tailcfg.Hostinfo) {
...
+// next update.
+func (c *Direct) SetNetInfo(ni *tailcfg.NetInfo) {
+	if ni == nil {
+		panic("nil NetInfo")
+	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.logf("NetInfo: %v\n", ni)
+	if c.hostinfo != nil {
```
Comment (crawshaw) [inline control/controlclient/direct.go]: Should we silently drop NetInfo here, or create a c.hostinfo?

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -268,6 +279,25 @@ type NetInfo struct {
 	// material change, as any change here also gets uploaded to
 	// the control plane.
 	DERPLatency map[string]float64 `json:",omitempty"`
+
+	// Update Clone and BasicallyEqual when adding fields.
+}
+
+// BasicallyEqual reports whether ni and ni2 are basically equal, ignoring
+// changes in DERPLatency.
+func (ni *NetInfo) BasicallyEqual(ni2 *NetInfo) bool {
```
Comment (crawshaw) [inline tailcfg/tailcfg.go]: Not edgy enough. How about:
- KindaEqual
- ApproxEqual
- CloseEnoughForNetworkSoftware

File: wgengine/wgengine.go [hunk 3]
```diff
@@ -40,6 +40,9 @@ type Status struct {
 // Exactly one of Status or error is non-nil.
 type StatusCallback func(*Status, error)
 
+// StatusCallback is the type used by Engine.SetNetInfoCallback.
```
Comment (crawshaw) [inline wgengine/wgengine.go]: out of date comment

File: control/controlclient/direct.go (hunk 1 above)
Comment (bradfitz) [reply control/controlclient/direct.go]: I added a log that it's being ignored instead. It really shouldn't be called with hostinfo nil.

Filed https://github.com/tailscale/tailscale/issues/136 instead to add a log convention for such things.

File: wgengine/wgengine.go (hunk 3 above)
Comment (bradfitz) [reply wgengine/wgengine.go]: done

**Pull request merged**